cmd/snap/cmd_quota.go

```diff
@@ -316,12 +424,45 @@ func (x *cmdQuotas) Execute(args []string) (err error) {
 			return fmt.Errorf("internal error: constraints is missing from daemon response")
 		}
 
-		constraintVal := "memory=" + strings.TrimSpace(fmtSize(int64(q.Constraints.Memory)))
-		currentVal := ""
-		if q.Current != nil && q.Current.Memory != 0 {
-			currentVal = "memory=" + strings.TrimSpace(fmtSize(int64(q.Current.Memory)))
+		var grpConstraints []string
+
+		// format memory constraint as memory=N
+		if q.Constraints.Memory != 0 {
+			grpConstraints = append(grpConstraints, "memory="+strings.TrimSpace(fmtSize(int64(q.Constraints.Memory))))
+		}
+
+		// format cpu constraint as cpu=NxM%,cpu-set=x,y,z
+		if q.Constraints.CPU != nil {
+			if q.Constraints.CPU.Count != 0 {
+				grpConstraints = append(grpConstraints, fmt.Sprintf("cpu=%dx", q.Constraints.CPU.Count))
+			}
+			if q.Constraints.CPU.Percentage != 0 {
+				grpConstraints = append(grpConstraints, fmt.Sprintf("cpu=%d%%", q.Constraints.CPU.Percentage))
+			}
+		}
+
+		if q.Constraints.CPUSet != nil && len(q.Constraints.CPUSet.CPUs) > 0 {
+			cpus := strutil.IntsToCommaSeparated(q.Constraints.CPUSet.CPUs)
+			grpConstraints = append(grpConstraints, "cpu-set="+cpus)
 		}
-		fmt.Fprintf(w, "%s\t%s\t%s\t%s\n", q.GroupName, q.Parent, constraintVal, currentVal)
+
+		// format threads constraint as thread=N
+		if q.Constraints.Threads != 0 {
+			grpConstraints = append(grpConstraints, "thread="+strconv.Itoa(q.Constraints.Threads))
```

Comment:
We don't seem to be hitting this case in the test.

Reply:
added now